File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/common/configuration/GetConfiguration.java
package com.crpr.androidcinema.domain.common.configuration;

import com.crpr.androidcinema.data.api.models.configuration.ApiConfiguration;
import com.crpr.androidcinema.domain.common.Result;

import rx.Observable;

/**
 * Created by claudioribeiro on 09/07/16.
 */
public interface GetConfiguration {

    interface Process {
        Observable<Result> getConfiguration();
    }

    interface Service {
        Observable<ApiConfiguration> getConfiguration();
    }
}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/common/configuration/GetConfiguration.kt
package com.crpr.androidcinema.domain.common.configuration

import com.crpr.androidcinema.data.api.models.ApiConfiguration
import com.crpr.androidcinema.domain.common.Result

import rx.Observable

/**
 * Created by claudioribeiro on 09/07/16.
 */
interface GetConfiguration {

    interface Process {
        val configuration: Observable<Result>
    }

    interface Service {
        val configuration: Observable<ApiConfiguration>
    }
}
